probe-rs/src/core/registers.rs

```diff
@@ -293,151 +351,220 @@ impl<T> RegisterValueResultExt<T> for Result<T, Infallible> {
+            Error::GenericCoreError(format!(
+                "Platform register {index:?} not found. Please report this as a bug."
+            ))
+        })
     }
 
     /// The frame pointer.
-    pub fn frame_pointer(&self) -> &RegisterDescription {
-        self.frame_pointer
+    pub fn frame_pointer(&self) -> Result<&CoreRegister, Error> {
+        self.0
+            .iter()
+            .find(|r| r.role == Some(RegisterRole::FramePointer))
+            .ok_or_else(|| {
+                Error::GenericCoreError(
+                    "No frame pointer found. Please report this as a bug.".to_string(),
+                )
+            })
     }
 
     /// The program counter.
-    pub fn program_counter(&self) -> &RegisterDescription {
-        self.program_counter
+    pub fn program_counter(&self) -> Result<&CoreRegister, Error> {
+        self.0
+            .iter()
+            .find(|r| r.role == Some(RegisterRole::ProgramCounter))
+            .ok_or_else(|| {
+                Error::GenericCoreError(
+                    "No program counter found. Please report this as a bug.".to_string(),
+                )
+            })
     }
 
     /// The stack pointer.
-    pub fn stack_pointer(&self) -> &RegisterDescription {
-        self.stack_pointer
+    pub fn stack_pointer(&self) -> Result<&CoreRegister, Error> {
+        self.0
+            .iter()
+            .find(|r| r.role == Some(RegisterRole::StackPointer))
+            .ok_or_else(|| {
+                Error::GenericCoreError(
+                    "No stack pointer found. Please report this as a bug.".to_string(),
+                )
+            })
     }
 
     /// The link register.
-    pub fn return_address(&self) -> &RegisterDescription {
-        self.return_address
+    pub fn return_address(&self) -> Result<&CoreRegister, Error> {
+        self.0
+            .iter()
+            .find(|r| r.role == Some(RegisterRole::ReturnAddress))
+            .ok_or_else(|| {
+                Error::GenericCoreError(
+                    "No return address found. Please report this as a bug.".to_string(),
+                )
+            })
     }
 
     /// Returns the nth argument register.
     ///
     /// # Panics
     ///
     /// Panics if the register at given index does not exist.
-    pub fn argument_register(&self, index: usize) -> &RegisterDescription {
-        &self.argument_registers[index]
+    pub fn argument_register(&self, index: usize) -> &CoreRegister {
+        self.get_argument_register(index).unwrap()
     }
 
     /// Returns the nth argument register if it is exists, `None` otherwise.
-    pub fn get_argument_register(&self, index: usize) -> Option<&RegisterDescription> {
-        self.argument_registers.get(index)
-    }
-
-    /// Returns the nth result register.
-    ///
-    /// # Panics
-    ///
-    /// Panics if the register at given index does not exist.
-    pub fn result_register(&self, index: usize) -> &RegisterDescription {
```

Comment:
This shouldn't be removed, the result registers are necessary for the flash loader. And they are not the same as the argument registers.

Reply:
Agreed. I will make the change and flip it back to you for review.

Reply:
Done